src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1284,16 +1284,13 @@
 (defn update-latest-report!
   "Given a node name, updates the `certnames` table to ensure that it indicates the
    most recent report for the node."
-  [node]
-  {:pre [(string? node)]}
-  (let [latest-report (:id (first (query-to-vec
-                                    ["SELECT id FROM reports
-                                      WHERE certname = ?
-                                      ORDER BY producer_timestamp DESC
-                                      LIMIT 1" node])))]
-    (jdbc/update! :certnames
-                  {:latest_report_id latest-report}
-                  ["certname = ?" node])))
+  [node report-id report-timestamp]
```

Comment:
should call this producer-timestamp instead to be as specific as possible (there are a few different timestamps on reports)